### DISCUSSION TITLE
Unable to Create Backup

### DISCUSSION THREAD

**Discussion by KennethS2022** [teslamate-org/teslamate#3128] [Q&A]
When I run the shell command to create a backup I get the following error:

-bash: docker-compose: command not found

I have checked and docker compose is installed and running the latest version. I have tried removing and re-installing the docker compose plug-in but still get the same error.  I am using a Raspberry Pi 4B with docker & portainer.  I am a novice at this (obviously) and any advice would be greatly appreciated.

**Comment by ktsenvolt** [comment]
You are on the wrong path.
sudo find / -name 'docker-compose' # use this command to find 'docker-compose'

**Reply by KennethS2022** [reply]
Thank you for the guidance. Much appreciated. Will try again tomorrow.

**Reply by KennethS2022** [reply]
Okay, I ran the command and got this output.  Any tips on what to do next?  Sorry for the novice questions!

/usr/libexec/docker/cli-plugins/docker-compose
find: ‘/run/user/1000/gvfs’: Permission denied
/var/lib/docker/overlay2/80fa1ca3136579dfe8edf6305be49edb9cb2ee115b0b1c9ee325ad49598575a9/merged/docker-compose
/var/lib/docker/overlay2/7149ff2860d71077939b36b8ce0ff8fa111fd39048ab784c13d84f43ea597b9a/diff/docker-compose

**Reply by ktsenvolt** [reply]
Can you share a screenshot when an issue occurs? For example : docker-compose exec -T database pg_dump -U teslamate teslamate >teslamate.bck

**Reply by KennethS2022** [reply]
I am clearly doing something supremely naive but I ran the command both from within the path that I believe the docker-compose service is located and also from within the TeslaMate docker container as well.  Both times it seemed to produce the same "command not found" error message.  I attach screenshots for both.  Thanks for the continued help!  I had hoped to be a quick learner but my age is to helping. I am enjoying TeslaMate but am not inclined to cancel TeslaFi until I have figured out how to make backups from TeslaMate. 

<img width="922" alt="Screenshot 2023-02-20 at 18 19 15" src="https://user-images.githubusercontent.com/122271978/220178123-5ae2bf4c-eabc-4168-8643-067f75efd1ce.png">
<img width="922" alt="Screenshot 2023-02-20 at 18 21 27" src="https://user-images.githubusercontent.com/122271978/220178141-c7655483-feb4-49d2-9da8-b2643576db4b.png">

**Reply by ktsenvolt** [reply]
Try changing user to root. That should take care of 'permission denied' problem

**Reply by KennethS2022** [reply]
Now I get the command not found error again. I sent another screenshot and also one showing the output from "docker ps" if that info is at all relevant. 

 
<img width="1032" alt="Screenshot 2023-02-20 at 19 58 30" src="https://user-images.githubusercontent.com/122271978/220191706-3025e3f0-ebb3-4c89-bcf9-217040b96cde.png">
<img width="1032" alt="Screenshot 2023-02-20 at 20 00 38" src="https://user-images.githubusercontent.com/122271978/220191730-a8058806-147f-40e0-8de2-d9cf829c628b.png">

**Reply by KennethS2022** [reply]
One more screenshot showing the output from find / -name 'docker-compose'

<img width="1032" alt="Screenshot 2023-02-20 at 20 21 13" src="https://user-images.githubusercontent.com/122271978/220192265-19fb62c9-b15a-4aad-bbd5-b7f5f0bfccc1.png">

**Reply by ktsenvolt** [reply]
https://forums.docker.com/t/command-not-found-when-i-try-to-run-docker-compose/97183

**Reply by KennethS2022** [reply]
I really appreciate all your help and time. I came to the conclusion that my setup must have been screwed up somehow so have nuked the whole thing and have started with a fresh installation of Raspberry Pi OS. Not sure if this will help but it is part of my learning process.  Thanks again!

**Reply by ktsenvolt** [reply]
![image](https://user-images.githubusercontent.com/80360674/220354580-723648f1-380a-48d7-8bb7-4b2b4412c4ae.png)
I used 'SD Card Copier' to make backup. It will copy the entire OS and data.
https://all3dp.com/2/back-up-raspberry-pi-sd-card/

**Reply by KennethS2022** [reply]
I considered making a duplicate of the storage drive but decided to just start over. I've made progress and have TeslaMate up and running again. I'm now able to run the docker-compose command but need to spend more time refining where I'm pointing the backup instruction since I'm ending up with a 0 byte file. But I'm feeling that progress has been made!

**Reply by KennethS2022** [reply]
Just to close this out, I have now been able to create a backup with the fresh installation. I have successfully imported 3+ years of data from TeslaFi into TeslaMate and all looks right.  Final question if I may: my teslamate.bck file is 19.2MB in size. Does that seem right?  I'm hesitant to attempt a restore from the file just to check.  I have also created a USB copy of the entire SD card in case I ever need it.  Thank you so much!

**Reply by ktsenvolt** [reply]
![image](https://user-images.githubusercontent.com/80360674/220631795-9d0bb88c-a5f2-4c2f-a0f0-9c5c08c5ac17.png)
Those were my 1.5 years data.

**Reply by KennethS2022** [reply]
Thanks, yes it confirms my fear. I think I'm getting closer but my backup file would have to be larger than what is output.  Will keep reading and learning....

**Comment by DrMichael** [comment]
Could you try `docker compose`?

**Reply by KennethS2022** [reply]
Thanks for the suggestion but still no joy.
